src/aaudio/AudioStreamAAudio.h

```diff
@@ -98,24 +98,53 @@ class AudioStreamAAudio : public AudioStream {
 
     void onErrorInThread(AAudioStream *stream, Result error);
 
-
-    void *getUnderlyingStream() const override {
-        return mAAudioStream.load();
-    }
+    bool usesMMap() override;
 
 protected:
     void updateFramesRead() override;
     void updateFramesWritten() override;
 
+    static AAudioLoader *mLibLoader;
+
 private:
 
-    bool                 isMMapUsed();
+    /** Define a class that wraps a 'C' style AAudioStream pointer.
+     * We can then use a std:shared_ptr to manage the ownership of that pointer.
+     */
+    class   AAudioStreamOwner {
+    public:
+
```

Comment:
I would suggest making this class non-copyable by deleting copy constructor and copy assignment.